### ISSUE TITLE
"File not found at path" when upgrading from 1.0.16

### ISSUE THREAD

**Issue by muhlegg** [thephpleague/flysystem#775]
Updated from 1.0.16 to newest version but ran into a strange issue. Read command throws "File not found at path: xxx" although the file exists. Tried 1.0.20 as well but same problem there. Reverted back to 16 and everything works fine again. 

Any ideas? Using standard FTP adapter. Listing and writing did work, only read that failed.

**Comment by RubtsovAV** [comment]
Can you write full name of your ftp server and example of path?

**Comment by RubtsovAV** [comment]
Do you try of the last version 1.0.37?

**Comment by muhlegg** [comment]
The FTP server is private and IP restricted. The path is just a regular _folder/folder/timestamped_file.xml_

Did try with the latest version.

**Comment by RubtsovAV** [comment]
We need the vendor name of ftp server and the exact path to repeat your issue.
The path have space symbols?
P.s. List of most popular ftp servers https://en.wikipedia.org/wiki/List_of_FTP_server_software

**Comment by muhlegg** [comment]
No space symbols used. 

The server is not hosted by me so I'll have to check with the hosting company what type of FTP they are running. I'll get back to you when i know the type..

**Comment by RubtsovAV** [comment]
You can try that for find it:
```
<?php
$fp = ftp_connect($host = "ftp.example.com", $port = 21);
$response = ftp_raw($fp, "HELP");
var_dump($response);
```
Most often, the response will be have the name and version of the FTP server.

**Comment by RubtsovAV** [comment]
Another way:
```
<?php
$socket = fsockopen($host = "ftp.example.com", $port = 21); 
echo fgets($socket); //read the welcome line
fclose($socket);
```

**Comment by vasekjares** [comment]
Same issue here. It seems that it's because of Ftp::getMetadata method.
`$listing = $this->ftpRawlist('-A', str_replace('*', '\\*', $path));`
Returns empty array. If I do:
`$this->filesystem->getConfig()->set('disable_asserts', true);`
Before read call, file is read correctly.

According to php doc: http://php.net/manual/en/function.ftp-rawlist.php second parameter should be directory and not actual filename.

I'm using latest flysystem and php7.1

**Comment by RubtsovAV** [comment]
Can you write of the example code, which not working? And, please, write your server vendor name (pureFTPd, vsftpd, ... ). So that we can repeat the issue.

I'm also using latest version and everything OK.

**Comment by vasekjares** [comment]
vendor/version: 

	vsftpd -v
	vsftpd: version 3.0.2 

example:

	<?php

	require('vendor/autoload.php');

	use League\Flysystem\Filesystem;
	use League\Flysystem\Adapter\Ftp as Adapter;

	$filesystem = new Filesystem(new Adapter([
	    'host' => $host,
	    'username' => $user,
	    'password' => $password,

	    /** optional config settings */
	    'port' => 21,
	    //'root' => '/path/to/root',
	    'passive' => true,
	    'ssl' => true,
	    'timeout' => 30,
	]));


	$contents = $filesystem->listContents('test');
	var_dump($contents);
	var_dump($filesystem->has($contents[0]['path']));


output:

    array(1) {
      [0]=>
      array(8) {
        ["type"]=>
        string(4) "file"
        ["path"]=>
        string(13) "test/test.txt"
        ["visibility"]=>
        string(6) "public"
        ["size"]=>
        int(62)
        ["dirname"]=>
        string(4) "test"
        ["basename"]=>
        string(8) "test.txt"
        ["extension"]=>
        string(3) "txt"
        ["filename"]=>
        string(4) "test"
      }
    }
    bool(false)

**Comment by vasekjares** [comment]
Let me know if you need further information. I can give you temporary access to the ftp. (via email or so)

**Comment by vasekjares** [comment]
(Ftpd adapter gives me the same result)

**Comment by RubtsovAV** [comment]
I was try this code and all works fine.
PHP 7.1.3
vsftpd 3.0.2 (on centos:7)

My Skype: RubtcovAV (please, write that you are from github)

**Comment by alexciarlillo** [comment]
I have just started running into this as well but I am using the SftpAdapter. Not sure how related the issue might be. For me the issue appears to be sporadic while downloading files from the remote server. I will have some go through fine and certain times get a ton of errors.

I have provided a sampling of the stack trace, directory listing with timestamps and sample calling code (laravel specific) here: https://github.com/thephpleague/flysystem-sftp/issues/67
